store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -10828,4 +10873,57 @@ public void noOp() throws ServiceException {
     public void resetDefaultCalendarId() throws ServiceException {
         getAccount().setPrefDefaultCalendarId(ID_FOLDER_CALENDAR);
     }
+
+    /**
+     * Method is used to call the API based on request being formed
+     * @param acct
+     * @param request
+     * @param uri
+     * @param authRequired
+     * @param timeout
+     * @return
+     * @throws ServiceException
+     */
+    public static Element sendRequest(Account acct, Element request, String uri, boolean authRequired, int timeout)
+            throws ServiceException {
+        assert request != null;
+        assert acct != null;
+
+        String soapUri = null;
+        if (StringUtil.isNullOrEmpty(uri)) {
+            soapUri = AccountUtil.getSoapUri(acct);
+        } else {
+            soapUri = uri + AccountConstants.USER_SERVICE_URI;
+        }
+
+        SoapHttpTransport transport = new SoapHttpTransport(soapUri);
+        try {
+            transport.setUserAgent(ZIMBRA_SOAP_USER_AGENT, ZIMBRA_SOAP_VERSION);
+            transport.setTimeout(timeout);
+            transport.setRetryCount(ZIMBRA_SOAP_RETRY_COUNT);
+            transport.setOriginalUserAgent("createDocumentRevision");
+            if (authRequired) {
+                try {
+                    transport.setAuthToken(AuthProvider.getAuthToken(acct).getEncoded());
+                } catch (AuthTokenException e) {
+                    throw ServiceException.FAILURE("Failed to generate auth token for " + acct.getName(), e);
+                }
+            }
+            transport.setRequestProtocol(SoapProtocol.Soap12);
+
+            printXml(request);
+            Element response = transport.invokeWithoutSession(request.detach());
+            printXml(response);
+
+            return response;
+        } catch (IOException e) {
+            throw ServiceException.PROXY_ERROR(e, soapUri);
+        } finally {
+            transport.shutdown();
+        }
+    }
+
+    public static void printXml(Element e) {
```

Comment:
I would call this method something `logDebugXML`

Reply:
renamed accordingly